velox/exec/HashJoinBridge.h

```diff
@@ -169,4 +169,13 @@ bool isHashBuildMemoryPool(const memory::MemoryPool& pool);
 /// Returns true if 'pool' is a hash probe operator's memory pool. The check is
 /// currently based on the pool name.
 bool isHashProbeMemoryPool(const memory::MemoryPool& pool);
+
+/// Returns the type used to spill a given hash table type. The function
+/// attaches a boolean column at the end of 'tableType'. It is used by the hash
+/// join table spilling triggered at the probe side to record if each row has
+/// been probed or not for join type like right join.
+RowTypePtr hashJoinTableSpillType(const RowTypePtr& tableType);
```

Comment:
Can we make sure to add an extra column only when needed, e.g. not add it for non-right joins.